backend/tests/end_to_end/infrastructures/apps/load/test_api.py

```diff
@@ -0,0 +1,452 @@
+        )
+        == data_2.id
+    )
+    assert (
+        json_response.get(infrastructure_common_consts.PAGINATION_PREVIOUS_LINK_NAME)
+        is not None
+    )
+    assert (
+        json_response.get(infrastructure_common_consts.PAGINATION_NEXT_LINK_NAME)
+        is None
+    )
+
+
+@pytest.mark.parametrize(
+    ("pagination_key", "pagination_value"),
+    (
+        (
+            infrastructure_common_consts.PAGINATION_OFFSET_QUERY_PARAMETER_NAME,
+            "invalid-offset",
+        ),
+        (
+            infrastructure_common_consts.PAGINATION_LIMIT_QUERY_PARAMETER_NAME,
+            "invalid-records_per_page",
+        ),
+    ),
+)
+def test_list_data_endpoint_handles_invalid_pagination_parameters(
+    unauthenticated_client: APIClientData, pagination_key: str, pagination_value: str
+):
+    # Given
+    client = unauthenticated_client.client
+
+    # When
+    response = client.get(
+        get_url(
+            path_name="load-list",
+            query_params={
+                pagination_key: pagination_value,
+            },
+        ),
+    )
+
+    # Then
+    assert response.status_code == HTTPStatus.BAD_REQUEST
+    assert infrastructure_common_consts.ERROR_DETAIL_KEY in response.data
+
+
+def test_list_data_endpoint_filtering_by_is_satisfied(
+    unauthenticated_client: APIClientData,
+):
+    # Given
+    client = unauthenticated_client.client
+
+    data = model_factories.DataFactory.create(data=fakers.fake_data(is_satisfied=True))
+    model_factories.DataFactory.create(data=fakers.fake_data(is_satisfied=False))
+
+    # When
+    response = client.get(
+        get_url(
+            path_name="load-list",
+            query_params={"is_satisfied": data.data["is_satisfied"]},
+        )
+    )
+
+    # Then
+    assert response.status_code == HTTPStatus.OK
+    assert response.data[infrastructure_common_consts.PAGINATION_COUNT_NAME] == 1
+    assert all(
+        item["is_satisfied"] == data.data["is_satisfied"]
+        for item in response.data[infrastructure_common_consts.PAGINATION_RESULTS_NAME]
+    )
+
+
+def test_list_data_endpoint_filtering_by_timestamp_to(
+    unauthenticated_client: APIClientData,
+):
+    # Given
+    client = unauthenticated_client.client
+    data = model_factories.DataFactory.create()
+    model_factories.DataFactory.create()
+
+    # When
+    response = client.get(
+        get_url(
+            path_name="load-list",
+            query_params={"timestamp_to": data.created_at.isoformat()},
+        )
+    )
+
+    # Then
+    assert response.status_code == HTTPStatus.OK
+    assert response.data[infrastructure_common_consts.PAGINATION_COUNT_NAME] == 1
+    assert all(
+        datetime.fromisoformat(item["timestamp"]) >= data.created_at
+        for item in response.data[infrastructure_common_consts.PAGINATION_RESULTS_NAME]
+    )
+
+
+def test_list_data_endpoint_filtering_by_timestamp_from(
```

Comment:
We are missing combined test filtering.

Reply:
added test_list_data_endpoint_filtering_by_multiple_criteria
please have a look

Reply:
please, rename it to a few attributes